### PR TITLE
fix(1308/edge-identity-pagination): misc pagination fixes

### LINKED ISSUE #1308: Enable Paging on Edge API Identities page

### PR DESCRIPTION
**Pull request by gagantrivedi**

### CODE REVIEW

File: api/app/pagination.py [hunk 1]
```diff
@@ -29,33 +76,37 @@ def paginate_queryset(self, dynamo_queryset, request, view=None):
...
+                        "last_evaluated_key",
+                        self.last_evaluated_key
+                        if hasattr(self, "last_evaluated_key")
+                        else None,
+                    ),
                 ]
             )
         )
+
+    def get_paginated_response_schema(self, schema):
```
Comment (matthewelwell) [inline api/app/pagination.py]: Seems odd that we need this as well as the `get_paginated_response` method on the Inspector class above?

File: api/tests/integration/edge_api/identities/test_edge_identity_viewset.py [hunk 2]
```diff
@@ -174,31 +174,34 @@ def test_identity_list_pagination(
...
+    # Make the call using the `last_evaluated_key`
+    url = f"{url}&last_evaluated_key={last_evaluated_key}"
+    response = admin_client.get(url)
 
     # And verify that get_all_items was called with correct arguments
     dynamo_wrapper_mock.get_all_items.assert_called_with(
         environment_api_key, 1, identity_item_key
     )
-    # And previous_url is same as last next_url
+    # And `last_evaluated_key` is now None
```
Comment (matthewelwell) [inline api/tests/integration/edge_api/identities/test_edge_identity_viewset.py:202]: So, wait, we're expecting `last_evaluated_key` to be `None` when the caller reaches the end of the results? That seems a bit counterintuitive but perhaps it's a dynamo thing? Feels like I'd expect e.g. ` next_key` to be `None` but surely `last_evaluated_key` would always be populated as long as there are results?

Comment (gagantrivedi) [reply api/tests/integration/edge_api/identities/test_edge_identity_viewset.py:202]: Yeah, it's a dynamo thing

> If there is not a LastEvaluatedKey in the result, there are no more items to be retrieved.
https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Query.Pagination.html

File: api/app/pagination.py (hunk 1 above)
Comment (gagantrivedi) [reply api/app/pagination.py]: Yeah, it's not useful anymore

**Review by matthewelwell — APPROVED** [review]

**Comment by gagantrivedi** [comment]
Holding on because of the prod build today

**Pull request merged**